src/thermostate/thermostate.py

```diff
@@ -309,6 +334,24 @@ def label(self, value: str | None):
                 f"The given label '{value!r}' could not be converted to a string"
             ) from None
         self._label = label
+        
+    @property
+    def units(self):
+        """Get or set the string label for this state, used in plotting."""
+        return self._units
+
+    @units.setter
+    def units(self, value: str | None):
+        if value is None:
+            self._units = value
+            return
+        try:
+            units = str(value)
+        except Exception:
+            raise TypeError(
+                f"The given label '{value!r}' could not be converted to a string"
+            ) from None
```

Comment:
I don't think you need to convert the value to a string here, it should either come in as a string `"SI"` or `"EE"`, or  `None`. I think the check you want to do here is whether the `value` is one of those values, like you did in `set_default_units()`.